pkg/profiles/tap.go

```diff
@@ -81,7 +84,8 @@ func routeSpecFromTap(tapByteStream *bufio.Reader, routeLimit int) []*sp.RouteSp
 		if err != nil {
 			// expected errors when hitting the tapDuration deadline
 			if err != io.EOF &&
-				!strings.HasSuffix(err.Error(), "(Client.Timeout exceeded while reading body)") &&
+				!strings.HasSuffix(err.Error(), "(Client.Timeout or context cancellation while reading body)") &&
```

Comment:
Can we use `errors.Is` here instead of string matching? https://golang.org/pkg/errors/#Is

Reply:
As we are wrapping the underlying error with others through the returns in the call stack, looks like we can't do this unless we used a separate error type that has this underlying i.e https://blog.golang.org/go1.13-errors

Reply:
I think we can fix this by using the %w for wrapped errors in `fmt.Errrorf`.  If that's easy to do it would be nice to clean that up now while we're in here.  Otherwise, I guess we can fix in a separate PR.

Reply:
Looks like not all error types used here, are exposed from go std library. :\

- https://sourcegraph.com/github.com/golang/go/-/blob/src/net/http/client.go#L723:19
- https://sourcegraph.com/github.com/golang/go/-/blob/src/net/http/h2_bundle.go#L8642

I was able to find only one instance here where this could be used but there are other places in other files where this could be done. I will raise an issue and fix this in a separate PR.

Reply:
I believe that this httpError implements the `net.Error` interface.  So I think we can use `errors.As` to unwrap the error into a `net.Error` and then check if it's a timeout with the `Timeout()` method on the `net.Error` interface.

Reply:
Hmm, We can do that. Riased https://github.com/linkerd/linkerd2/pull/5140 to work on this.